### DISCUSSION TITLE
How to configure delegate ps:alpha for AppImage on Linux?

### DISCUSSION THREAD

**Discussion by seal-pk** [ImageMagick/ImageMagick#7580] [Help] [answered]
### ImageMagick version

7.1.1-36

### Operating system, version and so on

Windows 10 x64, SLES 15 SP3 x64

### Description

We are using ImageMagick 7.1.1-36 Q16 x64 on Windows and the AppImage ImageMagick 7.1.1-36 Q16-HDRI on Linux to compare PDF- and PostScript documents for quality assurance.
For Windows we adjusted the delegates ps:alpha, ps:color and ps:cmyk in delegates.xml to call Ghostscript only with -dTextAlpaBits=1, -dGraphicsAlphaBits=1 and -sDEVICE=pnmraw.
These have been our settings since ImageMagick 6.7.7-6 2012-06-03 Q16. And we want to keep the settings to get the same MAE-values for magick compare as before. And it works fine on Windows.
With the AppImage on Linux, I am not able to adjust the Ghostscript calls. magick compare always calls gs with the -sDEVICE=pngalpha.
I can place my delegates.xml at ~/.config/ImageMagick and at $MAGICK_CONFIGURE_PATH. And I can see with 

strace -o strace.log magick compare -metric MAE -highlight-color green -lowlight-color black -compose Difference -colorspace RGB -negate -quality 90 file1.pdf[0] file2.pdf[0] compare_im7.jpg

where it searches and finds the delegates.xml.
The first one it finds is the internal version of the AppImage which gets mounted somehow at $TMPDIR like /tmp/.mount_magick20C4kV/usr/etc/ImageMagick-7/delegates.xml.
The second one is at $MAGICK_CONFIGURE_PATH. 
The third one is at ~/.config/ImageMagick. 
But magick seems to use the delegate ps:alpha only from the internal version. That's the only one where -sDEVICE=pngalpha is used.
Is there a way to tell the AppImage magick to use the delegates from the non-internal file?

Attached is a ZIP-file that contains our delegates.xml
[delegates.zip](https://github.com/user-attachments/files/16765800/delegates.zip)

**Comment by seal-pk** [comment, marked as answer]
My Solution for Linux: Replace gs by another program that manipulates the command line as needed. Then execute the original Ghostscript with the modified command line.

**Comment by snibgo** [comment]
You might consider creating your own pseudo-format called, for example, "pdfx". Add an entry to delegates.xml to decode this format with any command you wish, such as by calling Ghostscript with any arguments you want. In any `magick` command, prefix your input files with `pdfx:`.

**Reply by seal-pk** [reply]
Thank you for that hint! I placed my delegates.xml at $MAGICK_CONFIGURE_PATH and Ghostscript is called with the arguments that I specified. Now I have to go through our script where magick-commands are used to add the prefix to the file names.
Our script not just compares PostScript-files and PDF-files. But also TIFF-, PNG-, GIF-files. I have to take care to use the prefix only when PostScript or PDF is involved. For other file types, magick should select the correct delegate automatically.
